robbot.py

```diff
@@ -76,10 +77,25 @@ async def get_cocktail(msg):
         print(ex)
         await msg.channel.send('Ayo, your code is wack.')
 
-# TODO Find replyed message through author
-# Make emebd 
-# Post to Pin
+'''
+@author: Keeth S.
+@dependencies: util/Pin.py
+@desc: Sends a embed to the Pin channel when a user reply's to a message with .pin
+@retunrs: async message back to channel confirming message was pinned
+# TODO Optimize Drink Object.
+'''
 async def pin_message(msg):
-    pass
+    try:
+        if not msg.reference:
+            await msg.channel.send('Sorry, bud. Just can\'t do it.')
+            return
+        reply = await msg.channel.fetch_message(msg.reference.message_id)
+        pin_embed = Pin(reply)
+        pin_channel = client.get_channel(789771971532947486)
```

Comment:
Should avoid magic numbers - better would be to give the channel ID here a name (pin_channel_id = ...), better still would be to make this value configurable. The 8ball stuff adds a config file which we can add this value to once I merge it.